### PR TITLE
API for lifecycle associated with organization verification

### PR DESCRIPTION
**Pull request by techsmyth**
New entity for managing the verification process for organizations.

Lifecycle defined + assocated with this process
- Two different guards: one for updating the state (UPDATE) and one for assigning the verified status (GRANT). 
- Authorization policy is setup so that only global admins have the GRANT privilege on the verification entity

Migration is re-done to be after the organiZation renaming

This should be ready for a review once the organiZation PR is merged in. It will also require a client-web update as the status field for verification has been moved. 

NOte: need to reset authorization policy on orgs after doing the migration + running the server

### CODE REVIEW

File: src/domain/community/organization-verification/organization.verification.entity.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { Column, Entity, JoinColumn, OneToOne } from 'typeorm';
+import { IOrganizationVerification } from './organization.verification.interface';
+import { OrganizationVerificationEnum } from '@common/enums/organization.verification';
+import { Lifecycle } from '@domain/common/lifecycle/lifecycle.entity';
+import { AuthorizableEntity } from '@domain/common/entity/authorizable-entity';
+
+@Entity('OrganizationVerification')
```
Comment (valentinyanakiev) [inline src/domain/community/organization-verification/organization.verification.entity.ts]: Specifying the name of the entity as a string when the class name is the same is redundant - remove it.

File: src/migrations/1631992797354-organization-verification.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    );
+    await queryRunner.query(
+      'ALTER TABLE `OrganizationVerification` ADD CONSTRAINT `FK_22be0d440df7972d9b3a94aa6d5` FOREIGN KEY (`lifecycleId`) REFERENCES `lifecycle`(`id`) ON DELETE SET NULL ON UPDATE NO ACTION'
+    );
+    await queryRunner.query(
+      'ALTER TABLE `organization` ADD CONSTRAINT `FK_95bbac07221e98072beafa61732` FOREIGN KEY (`verificationId`) REFERENCES `OrganizationVerification`(`id`) ON DELETE SET NULL ON UPDATE NO ACTION'
+    );
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
```
Comment (valentinyanakiev) [inline src/migrations/1631992797354-organization-verification.ts:35]: That migration doesn't work:
> ts-node -r tsconfig-paths/register ./node_modules/typeorm/cli.js --config src/config/typeorm.cli.config.ts "migration:revert"

query: SELECT * FROM `INFORMATION_SCHEMA`.`COLUMNS` WHERE `TABLE_SCHEMA` = 'alkemio' AND `TABLE_NAME` = 'migrations_typeorm'
query: SELECT * FROM `alkemio`.`migrations_typeorm` `migrations_typeorm` ORDER BY `id` DESC
7 migrations are already loaded in the database.
organizationVerification1631992797354 is the last executed migration. It was executed on Sat Sep 18 2021 22:19:57 GMT+0300 (Eastern European Summer Time).
Now reverting it...
query: START TRANSACTION
query: ALTER TABLE `OrganizationVerification` DROP FOREIGN KEY `FK_22be0d440df7972d9b3a94aa6d5`
query: ALTER TABLE `OrganizationVerification` DROP FOREIGN KEY `FK_3795f9dd15ef3ef2dd1d27e309c`
query: ALTER TABLE `organization` DROP COLUMN `verificationId`
query failed: ALTER TABLE `organization` DROP COLUMN `verificationId`
error: Error: ER_FK_COLUMN_CANNOT_DROP: Cannot drop column 'verificationId': needed in a foreign key constraint 'FK_95bbac07221e98072beafa61732'
    at Query.Sequence._packetToError (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
    at Query.ErrorPacket (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/sequences/Query.js:79:18)
    at Protocol._parsePacket (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Protocol.js:291:23)
    at Parser._parsePacket (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Parser.js:433:10)
    at Parser.write (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Parser.js:43:10)
    at Protocol.write (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Protocol.js:38:16)
    at Socket.<anonymous> (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/Connection.js:88:28)
    at Socket.<anonymous> (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/Connection.js:526:10)
    at Socket.emit (events.js:375:28)
    at Socket.emit (domain.js:470:12)
    --------------------
    at Protocol._enqueue (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Protocol.js:144:48)
    at PoolConnection.query (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/Connection.js:198:25)
    at MysqlQueryRunner.<anonymous> (/home/valentin/source/repos/alkemio/server/src/driver/mysql/MysqlQueryRunner.ts:184:36)
    at step (/home/valentin/source/repos/alkemio/server/node_modules/tslib/tslib.js:143:27)
    at Object.next (/home/valentin/source/repos/alkemio/server/node_modules/tslib/tslib.js:124:57)
    at fulfilled (/home/valentin/source/repos/alkemio/server/node_modules/tslib/tslib.js:114:62)
    at processTicksAndRejections (internal/process/task_queues.js:95:5) {
  code: 'ER_FK_COLUMN_CANNOT_DROP',
  errno: 1828,
  sqlMessage: "Cannot drop column 'verificationId': needed in a foreign key constraint 'FK_95bbac07221e98072beafa61732'",
  sqlState: 'HY000',
  index: 0,
  sql: 'ALTER TABLE `organization` DROP COLUMN `verificationId`'
}
query: ROLLBACK
Error during migration revert:
QueryFailedError: ER_FK_COLUMN_CANNOT_DROP: Cannot drop column 'verificationId': needed in a foreign key constraint 'FK_95bbac07221e98072beafa61732'
    at QueryFailedError.TypeORMError [as constructor] (/home/valentin/source/repos/alkemio/server/src/error/TypeORMError.ts:7:9)
    at new QueryFailedError (/home/valentin/source/repos/alkemio/server/src/error/QueryFailedError.ts:9:9)
    at Query.<anonymous> (/home/valentin/source/repos/alkemio/server/src/driver/mysql/MysqlQueryRunner.ts:195:37)
    at Query.<anonymous> (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/Connection.js:526:10)
    at Query._callback (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/Connection.js:488:16)
    at Query.Sequence.end (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/sequences/Sequence.js:83:24)
    at Query.ErrorPacket (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/sequences/Query.js:92:8)
    at Protocol._parsePacket (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Protocol.js:291:23)
    at Parser._parsePacket (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Parser.js:433:10)
    at Parser.write (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Parser.js:43:10) {
  query: 'ALTER TABLE `organization` DROP COLUMN `verificationId`',
  parameters: undefined,
  driverError: Error: ER_FK_COLUMN_CANNOT_DROP: Cannot drop column 'verificationId': needed in a foreign key constraint 'FK_95bbac07221e98072beafa61732'
      at Query.Sequence._packetToError (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
      at Query.ErrorPacket (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/sequences/Query.js:79:18)
      at Protocol._parsePacket (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Protocol.js:291:23)
      at Parser._parsePacket (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Parser.js:433:10)
      at Parser.write (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Parser.js:43:10)
      at Protocol.write (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Protocol.js:38:16)
      at Socket.<anonymous> (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/Connection.js:88:28)
      at Socket.<anonymous> (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/Connection.js:526:10)
      at Socket.emit (events.js:375:28)
      at Socket.emit (domain.js:470:12)
      --------------------
      at Protocol._enqueue (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Protocol.js:144:48)
      at PoolConnection.query (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/Connection.js:198:25)
      at MysqlQueryRunner.<anonymous> (/home/valentin/source/repos/alkemio/server/src/driver/mysql/MysqlQueryRunner.ts:184:36)
      at step (/home/valentin/source/repos/alkemio/server/node_modules/tslib/tslib.js:143:27)
      at Object.next (/home/valentin/source/repos/alkemio/server/node_modules/tslib/tslib.js:124:57)
      at fulfilled (/home/valentin/source/repos/alkemio/server/node_modules/tslib/tslib.js:114:62)
      at processTicksAndRejections (internal/process/task_queues.js:95:5) {
    code: 'ER_FK_COLUMN_CANNOT_DROP',
    errno: 1828,
    sqlMessage: "Cannot drop column 'verificationId': needed in a foreign key constraint 'FK_95bbac07221e98072beafa61732'",
    sqlState: 'HY000',
    index: 0,
    sql: 'ALTER TABLE `organization` DROP COLUMN `verificationId`'
  },
  code: 'ER_FK_COLUMN_CANNOT_DROP',
  errno: 1828,
  sqlMessage: "Cannot drop column 'verificationId': needed in a foreign key constraint 'FK_95bbac07221e98072beafa61732'",
  sqlState: 'HY000',
  index: 0,
  sql: 'ALTER TABLE `organization` DROP COLUMN `verificationId`'
}

**Review by valentinyanakiev — CHANGES_REQUESTED** [review]
Fix down migration - currently it fails.

**Review by valentinyanakiev — APPROVED** [review]

**Review by hero101 — APPROVED** [review]

**Comment by techsmyth** [comment]
Checked out develop, ran migrations, started server so that default host org is created. Checked out this branch, ran migration, started - all fine. Also reverted - all fine. Good to go imho :)

**Pull request merged**